scene/gui/popup_menu.cpp

```diff
@@ -238,32 +238,64 @@ void PopupMenu::_gui_input(const Ref<InputEvent> &p_event) {
 			search_from = 0;
 		}
 
+		bool match_found = false;
 		for (int i = search_from; i < items.size(); i++) {
 			if (!items[i].separator && !items[i].disabled) {
 				mouse_over = i;
 				emit_signal("id_focused", i);
 				_scroll_to_item(i);
 				control->update();
 				set_input_as_handled();
+				match_found = true;
 				break;
 			}
 		}
+
+		if (!match_found) {
+			// If the last item is not selectable, try re-searching from the start.
```

Comment:
What if the second-to-last item is also not selectable? Don't you run into the same issue again?

Reply:
Ah nevermind, the two loops have a different starting point.